### ISSUE TITLE
indexing an array with decreasing time returns array with time dimension even for a single time selected

### ISSUE THREAD

**Issue by snowzky** [pydata/xarray#5277]
I believe I have found an additional bug related to non-increasing time (whether it needs to be monotonic I do not know).

Background: My own research data has decreasing time as standard output (back trajectories), but I will present this example using xr's tutorial data. I already found the bug that you can't select a forward going time slice in a backwards going data set. (I can't find the bug report anymore, sorry).

I know the work-around is easy, and it can be discussed if decreasing slices with labelled dimensions (when using sel) should support slices in either direction.
But that a _single_ selection still returns a time dimension surely is unintended.

**What happened**:
When selecting a single time step in a decreasing DataArray, the returned DataArray still has a time dimension, of size one.
This makes trouble for different situations that expects 2D data: for instance plot.contourf() which does not reduce the time dimension - as apparently .plot() does.

**What you expected to happen**:
A returned DataArray with no time dimension.

**Minimal Complete Verifiable Example**:

```python
ds = xr.tutorial.open_dataset("air_temperature.nc")
print(ds.air.sel(time='2014-12-31T18').shape)
>> (25, 53)
ds=ds.reindex(time=list(reversed(ds.time.values))) #reverse time to show effect
print(ds.air.sel(time='2014-12-31T18').shape)
>> (1, 25, 53)
```

**Environment**:

<details><summary>Output of <tt>xr.show_versions()</tt></summary>

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.9 (default, Aug 31 2020, 12:42:55) 
[GCC 7.3.0]
python-bits: 64
OS: Linux
OS-release: 4.15.0-91-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8
libhdf5: 1.10.4
libnetcdf: 4.7.3

xarray: 0.16.1
pandas: 1.1.3
numpy: 1.19.1
scipy: 1.5.2
netCDF4: 1.5.3
pydap: None
h5netcdf: None
h5py: None
Nio: None
zarr: None
cftime: 1.2.1
nc_time_axis: None
PseudoNetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: None
dask: None
distributed: None
matplotlib: 3.3.1
cartopy: 0.16.0
seaborn: 0.11.0
numbagg: None
pint: None
setuptools: 50.3.0.post20201006
pip: 20.2.3
conda: None
pytest: None
IPython: 7.18.1
sphinx: 3.2.1

</details>

**Comment by keewis** [comment]
I agree, this seems like a bug. Unfortunately, this is something we inherit from `pandas.DatetimeIndex`:
```python
In [2]: ds = xr.tutorial.open_dataset("air_temperature")
   ...: index = ds.indexes["time"]
   ...: display(index)
   ...: display(index.get_loc("2014-12-31T18"))
   ...: display(index[::-1].get_loc("2014-12-31T18"))
DatetimeIndex(['2013-01-01 00:00:00', '2013-01-01 06:00:00',
               '2013-01-01 12:00:00', '2013-01-01 18:00:00',
               '2013-01-02 00:00:00', '2013-01-02 06:00:00',
               '2013-01-02 12:00:00', '2013-01-02 18:00:00',
               '2013-01-03 00:00:00', '2013-01-03 06:00:00',
               ...
               '2014-12-29 12:00:00', '2014-12-29 18:00:00',
               '2014-12-30 00:00:00', '2014-12-30 06:00:00',
               '2014-12-30 12:00:00', '2014-12-30 18:00:00',
               '2014-12-31 00:00:00', '2014-12-31 06:00:00',
               '2014-12-31 12:00:00', '2014-12-31 18:00:00'],
              dtype='datetime64[ns]', name='time', length=2920, freq=None)
2919
array([0])
```
---
For reference, instead of
```python
ds.reindex(time=list(reversed(ds.time.values)))
```
you can use
```python
ds.isel(time=slice(None, None, -1))
```

**Comment by max-sixty** [comment]
Good find @keewis !

@snowzky would you be up for filing an issue upstream (or perhaps one already exists)?
